rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf

```diff
@@ -583,10 +592,53 @@ SecRule ARGS_NAMES|ARGS|XML:/* "@rx (?i)[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?(?:
     tag:'paranoia-level/2',\
     ver:'OWASP_CRS/3.4.0-dev',\
     severity:'CRITICAL',\
-    multiMatch,\
-    setvar:'tx.sql_injection_score=+%{tx.critical_anomaly_score}',\
-    setvar:'tx.anomaly_score_pl2=+%{tx.critical_anomaly_score}'"
+    setvar:'tx.942130_lhs=/%{TX.1}/',\
+    setvar:'tx.942130_matched_var_name=%{matched_var_name}',\
+    chain"
+    SecRule TX:942130_lhs "@within /%{TX.2}/" \
+        "t:none,\
+        setvar:'tx.sql_injection_score=+%{tx.critical_anomaly_score}',\
+        setvar:'tx.anomaly_score_pl2=+%{tx.critical_anomaly_score}'"
 
+# Rule Targeting logical inequalities that return TRUE (e.g. 1 != 2)
+#
+# The regexp written at the initial rule on the chain is the original prefix from the 942130.data.
+# When the prefix matches, we store the match to a new variable for comparing in the chained rule.
+#
+# We use captures to check for (in)equality in the regexp. So TX.1 will capture the left and side of the inequality,
+# and TX.2, TX.3 and TX.4 will capture the right hand side of the logical query.
+#
+# To regenerate the regexp:
+#
+#   cd util/regexp-assemble
+#   ./regexp-assemble.py data/942131.data
+#
+SecRule ARGS_NAMES|ARGS|XML:/* "@rx (?i)[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?(?:not\s+(?:r(?:egexp|like)|like)|is\s+not|![<=>]|<[=>]?|>=?|\^)[\s'\"`()]*?\b([\d\w]+)\b" \
+    "id:942131,\
+    phase:2,\
+    block,\
+    capture,\
+    t:none,t:urlDecodeUni,t:replaceComments,\
+    msg:'SQL Injection Attack: SQL Boolean-based attack detected',\
+    logdata:'Matched Data: %{TX.0} found within %{TX.942131_MATCHED_VAR_NAME}: %{MATCHED_VAR}',\
+    tag:'application-multi',\
+    tag:'language-multi',\
+    tag:'platform-multi',\
+    tag:'attack-sqli',\
+    tag:'OWASP_CRS',\
+    tag:'capec/1000/152/248/66',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/2',\
+    ver:'OWASP_CRS/3.4.0-dev',\
+    severity:'CRITICAL',\
+    multiMatch,\
+    setvar:'tx.942131_lhs=/%{TX.1}/',\
+    setvar:'tx.942131_matched_var_name=%{matched_var_name}',\
+    chain"
+    SecRule TX:942131_lhs "!@within /%{TX.2}/" \
```

Comment:
Since we compare a single value now, we could just use `@eq` and remove the delimiters from the variable and the operator.

Reply:
Agreed.